extra_data/calibration.py

```diff
@@ -0,0 +1,1090 @@
+            client (CalibrationClient, optional): Client for CalCat
+                communication, global one by default.
+            event_at (datetime, date, str or None): Default time at which the
+                CCVs should have been valid, now if omitted
+            snapshot_at (datetime, date, str or None): Default time of
+                database state to look at, now if omitted.
+            **condition_params: Operating condition parameters defined
+                on an instance level.
+        """
+
+        self.detector_name = detector_name
+        self.modules = modules
+        self.event_at = event_at
+        self.snapshot_at = snapshot_at
+
+        if client is None:
+            if self.__class__.default_client is None:
+                raise ValueError(f'need client passed by argument or prior '
+                                 f'client set-up via {self.__class__.__name__}'
+                                 f'.new_client')
+
+            client = self.__class__.default_client
+
+        self._api = CalCatApi(client)
+
+    @staticmethod
+    def new_client(client_id, client_secret, user_email, installation='',
+                   base_url='https://in.xfel.eu/{}calibration'):
+        """Create a new calibration-client object.
+
+        The client object is saved as a class property and is
+        automatically to any future CorrectionData objects created, if
+        no other client is passed explicitly.
+
+        Arguments:
+            client_id (str): Client ID.
+            client_secret (str): Client secret.
+            user_email (str): LDAP user email.
+            installation (str, optional): Prefix for CalCat
+                installation, production system by default.
+            base_url (str, optional): URL template for CalCat
+                installation, public European XFEL by default.
+
+        Returns:
+            (CalibrationClient) CalCat client.
+        """
+
+        base_url = base_url.format(f'{installation}_' if installation else '')
+        CorrectionData.default_client = CalibrationClient(
+            client_id=client_id,
+            client_secret=client_secret,
+            user_email=user_email,
+            base_api_url=f'{base_url}/api/',
+            token_url=f'{base_url}/oauth/token',
+            refresh_url=f'{base_url}/oauth/token',
+            auth_url=f'{base_url}/oauth/authorize',
+            scope='',
+        )
+        return CorrectionData.default_client
+
+    @property
+    def caldb_root(self):
+        """Root directory for calibration constant data.
+
+        Returns:
+            (Path or None) Location of caldb store or
+                None if not available.
+        """
+
+        if not hasattr(CorrectionData, '_caldb_root'):
+            if getenv('SASE'):
+                # ONC
+                CorrectionData._caldb_root = Path('/common/cal/caldb_store')
+            elif re.match(r'^max-(.+)\.desy\.de$', socket.getfqdn()):
```

Comment:
I'd be inclined to just check if the expected directories exist, rather than looking at environment variables and host names. E.g. inside a container, we may have a different hostname, even if the `/gpfs` paths are accessible.

I suspect we'll also want a way to override it, e.g. to use the test CalCat files at `/gpfs/exfel/d/cal_tst/caldb_store` on Maxwell.

Reply:
Very true, thanks for spotting the use case of test data.